app/src/lib/stores/app-store.ts

```diff
@@ -2913,6 +2924,7 @@ export class AppStore extends TypedBaseStore<IAppState> {
               repository,
               foundBranch.name
             )
+            this.statsStore.recordChangesTakenToNewBranch()
```

Comment:
i think this metric recording should be moved down to the `if` statement at L2962 (https://github.com/desktop/desktop/pull/7257/files#diff-d6b76b602a5c48740d6d10deda6f71ebR2962) since `stashToPop` can be set in places besides this and is actually popped later.

Reply:
I think you're right.  At the moment, it's possible to increment this metric even if checkout fails. I think this will get fixed once we refactor this flow to separate checking a branch out from stashing changes.

Reply:
f317219 and 4743c6a